src/BloomExe/Browser.cs

```diff
@@ -419,21 +419,23 @@ private void PasteFilteredText(bool removeSingleLineBreaks)
 
 		void OnShowContextMenu(object sender, GeckoContextMenuEventArgs e)
 		{
+			MenuItem FFMenuItem = null;
 			Debug.Assert(!InvokeRequired);
 			if (ContextMenuProvider != null)
 			{
-				ContextMenuProvider(e);
-#if DEBUG
-				e.ContextMenu.MenuItems.Add("Open Page in Firefox (which must be in the PATH environment variable)", new EventHandler(OnOpenPageInSystemBrowser));
-#endif
-				if (ContextMenuProvider(e))
-					return; // only the provider's items
+				var replacesStdMenu = ContextMenuProvider(e);
+				FFMenuItem = e.ContextMenu.MenuItems.Add("Open Page in Firefox (which must be in the PATH environment variable)",
```

Comment:
I guess I'm not really clear on when you might not want this (or when ContextMenuProvider might return 'true'). JH says he uses this in Release builds to check things out.